examples/Model.elm

```diff
@@ -81,90 +111,136 @@ update msg model =
+            , fetchRecordList model
+            )
+
+        CreateRecordResponse (Err error) ->
+            model |> updateError error
 
         EditRecord recordId ->
             ( model, fetchRecord recordId )
 
-        EditRecordResponse response ->
-            case response of
-                Ok _ ->
-                    ( model, fetchRecordList )
+        EditRecordResponse (Ok _) ->
+            ( model, fetchRecordList model )
 
-                Err err ->
-                    ( { model | error = Just <| toString err }, Cmd.none )
+        EditRecordResponse (Err error) ->
+            model |> updateError error
 
         DeleteRecord recordId ->
             ( model, deleteRecord recordId )
 
-        DeleteRecordResponse response ->
-            case response of
-                Ok record ->
-                    ( { model
-                        | records = removeRecordFromList record model.records
-                        , error = Nothing
-                      }
-                    , fetchRecordList
-                    )
+        DeleteRecordResponse (Ok record) ->
+            ( { model
+                | records = removeRecordFromList record model.records
+                , error = Nothing
+              }
+            , fetchRecordList model
+            )
+
+        DeleteRecordResponse (Err error) ->
+            model |> updateError error
+
+        UpdateFormTitle title ->
+            let
+                formData =
+                    model.formData
+
+                updated =
+                    { formData | title = title }
+            in
+                ( { model
+                    | formData = updated
+                    , records = updateRecordInList updated model.records
+                  }
+                , Cmd.none
+                )
+
+        UpdateFormDescription description ->
+            let
+                formData =
+                    model.formData
+
+                updated =
+                    { formData | description = description }
+            in
+                ( { model
+                    | formData = updated
+                    , records = updateRecordInList updated model.records
+                  }
+                , Cmd.none
+                )
+
+        Submit ->
+            ( { model | formData = initialFormData }, sendFormData model.formData )
+
+        SortByColumn column ->
+            let
+                sort =
+                    case model.sort of
+                        Asc sortedColumn ->
+                            if sortedColumn == column then
+                                (Desc sortedColumn)
+                            else
+                                (Asc column)
+
+                        Desc sortedColumn ->
+                            if sortedColumn == column then
+                                (Asc sortedColumn)
+                            else
+                                (Asc column)
```

Comment:
This functionality seems kind of complex and maybe a good candidate for testing of some kind. Also, I count 3 `Asc` and 1 `Desc`, is that right?